### ISSUE TITLE
Drush Composer Security Update command does not provide information on contributed vulnerabilities

### ISSUE THREAD

**Issue by generalconsensus** [drush-ops/drush#3731]
**Describe the bug**
The drush sec command does no correctly show the security status of contributed modules for Drupal 8 while using the Drush 9 version

**To Reproduce**
```
composer require 'drupal/entity_delete:^1.3' (https://www.drupal.org/sa-contrib-2018-040)

drush en entity_delete -y

docker@cli:/var/www/drupal/docroot$ drush sec
 [warning] One or more of your dependencies has an outstanding security update. Please apply update(s) immediately.
 [notice] Try running: composer require drupal/core:^8.5.6 --update-with-dependencies
 [notice] If that fails due to a conflict then you must update one or more root dependencies.
+-------------+-------------------+-------------------+
| Name        | Installed Version | Suggested version |
+-------------+-------------------+-------------------+
| drupal/core | 8.5.4             | 8.5.6             |
+-------------+-------------------+-------------------+

```
**Expected behavior**
I expected that Drush would provide information on this vulnerability 

**Actual behavior**
Drush did not provide the information on the contributed module vulnerability but only the core module vulnerability

**Workaround**
Not presently.

### System Configuration
| Q               | A
| --------------- | ---
| Drush version?  | 9.30
| Drupal version? | 8.5.4
| PHP version     | 7.1.20
| OS?             | Linux

**Additional information**

**Comment by generalconsensus** [comment]
I did some testing and it appears that the reasoning for this happening is that the Comparator dependency library correctly does not flag the insecure modules because of the way that the https://raw.githubusercontent.com/drupal-composer/drupal-security-advisories/8.x/composer.json formats the contributed modules. 

Example composer.lock showing alinks version stored:

```
{
            "name": "drupal/alinks",
            "version": "1.1.0",
            "source": {
                "type": "git",
                "url": "https://git.drupal.org/project/alinks",
                "reference": "8.x-1.1"
            },
            "dist": {
                "type": "zip",
                "url": "https://ftp.drupal.org/files/projects/alinks-8.x-1.1.zip",
                "reference": "8.x-1.1",
                "shasum": "6a3d26bf1109c25fc555c8535e062c496600f117"
            },
            "require": {
                "drupal/core": "*",
                "wamania/php-stemmer": "^1.0"
            },
            "type": "drupal-module",
            "extra": {
                "branch-alias": {
                    "dev-1.x": "1.x-dev"
                },
                "drupal": {
                    "version": "8.x-1.1",
                    "datestamp": "1501766275",
                    "security-coverage": {
                        "status": "covered",
                        "message": "Covered by Drupal's security advisory policy"
                    }
                }
            },
            "notification-url": "https://packages.drupal.org/8/downloads",
            "license": [
                "GPL-2.0+"
            ],
            "authors": [
                {
                    "name": "Greg Boggs",
                    "homepage": "https://www.drupal.org/user/153069"
                },
                {
                    "name": "Ivo.Radulovski",
                    "homepage": "https://www.drupal.org/user/337498"
                },
                {
                    "name": "chinajason",
                    "homepage": "https://www.drupal.org/user/276946"
                },
                {
                    "name": "gaurav.goyal",
                    "homepage": "https://www.drupal.org/user/2497208"
                },
                {
                    "name": "joebachana",
                    "homepage": "https://www.drupal.org/user/113430"
                },
                {
                    "name": "rsvelko",
                    "homepage": "https://www.drupal.org/user/337401"
                },
                {
                    "name": "tic2000",
                    "homepage": "https://www.drupal.org/user/103877"
                }
            ],
            "description": "Allows editors to automatically transform keyword phrases into links",
            "homepage": "https://www.drupal.org/project/alinks",
            "keywords": [
                "Drupal"
            ],
            "support": {
                "source": "http://cgit.drupalcode.org/alinks",
                "issues": "https://www.drupal.org/project/issues/alinks"
            }
        }
```

https://raw.githubusercontent.com/drupal-composer/drupal-security-advisories/8.x/composer.json :

```
{
"name": "drupal-composer/drupal-security-advisories",
"description": "Prevents installation of composer packages with known security vulnerabilities",
"type": "metapackage",
"license": "GPL-2.0-or-later",
"conflict": {
"drupal/alinks": "<1.1",
},
}
```

https://github.com/drush-ops/drush/blob/9.3.0/src/Commands/pm/SecurityUpdateCommands.php#L74 :

The variables I see are:
$package['version'] = 1.1.0;
$min_version = 1.1;

```
                        if (Comparator::lessThan($package['version'], $min_version)) {
                            $this->securityUpdates[$name] = [
                                'name' => $name,
                                'version' => $package['version'],
                                'min-version' => $min_version,
                            ];
                        }

```
So it fails.

When it succeeds with Drupal Core the values I have are:

$package['version'] = 8.5.4;
$min_version = 8.5.6;

**Comment by weitzman** [comment]
Any chance you can make a PR. You could add a test which catches the problem or identify such a test and someone else will write it.

**Comment by weitzman** [comment]
I tried your example with core and with entity_delete and I  am getting both being reported (i.e. correct behavior). 

```
+----------------------+-------------------+-------------------+
| Name                 | Installed Version | Suggested version |
+----------------------+-------------------+-------------------+
| drupal/core          | 8.5.8             | 8.6.2             |
| drupal/entity_delete | 1.3.0             | 1.4               |
+----------------------+-------------------+-------------------+
```

**Comment by shaysmith** [comment]
Just noting that I've encountered the same issue. Drush reports no outstanding security updates:
`$ drush pm:security
 [success] There are no outstanding security updates for Drupal projects.`

Yet if I use composer outdated and confirm on drupal.org I found that search_api_solr had an outstanding security update: 

`$ composer outdated --direct
    1/1:	https://packages.drupal.org/8/drupal/provider-2018-3$3438012a8afba3fb1c4028c8ccd78e48190ee9b415f2fb303089560ca1592548.json
    Finished: success: 1, skipped: 0, failure: 0, total: 1
    1/2:	http://repo.packagist.org/p/provider-latest$313933597667bb68d2700f438c769c633cb2435225c442a5c8b14c7b988aab40.json
    2/2:	http://repo.packagist.org/p/provider-2018-01$7498a8e8e39a6b040a9fd05310b3468cc5f6a5b3e18fe910c217c3cbee7c4f4e.json
    Finished: success: 2, skipped: 0, failure: 0, total: 2
drupal/better_exposed_filters 3.0.0-alpha4 3.0.0-alpha5 Provides advanced options (such as links, checkboxes, or j...
drupal/search_api_solr        1.2.0        2.2.0        Offers an implementation of the Search API that uses an Ap...
drupal/structure_sync         1.13.0       1.14.0       Tool for syncing structural data that is stored as content.
drupal/twig_tweak             2.0.0-rc4    2.1.0        A Twig extension with some useful functions and filters fo...`

**Comment by weitzman** [comment]
Could you guys retry with Drush's dev-master version? We fixed a bug at #3766 which may be at play here.

**Comment by generalconsensus** [comment]
@weitzman Be advised using your  Drush version : 9.6.0-beta3 version it appears fixed!

Thank you!
